File: lua/plugin/formatter.lua
return {
    {
        "stevearc/conform.nvim",
        opts = {
            formatters_by_ft = {
                lua = { "stylua" },
                python = { "black" },
                javascript = { "prettier" },
                typescript = { "prettier" },
                json = { "prettier" },
                markdown = { "prettier" },
                rust = { "rustfmt" },
                uiua = { "uiuafmt" },
            },
            formatters = {
                uiuafmt = {
                    command = "uiua",
                    args = { "fmt", "$FILENAME" },
                    stdin = false,
                },
            },
        },
        lazy = true,
    },
}

File: src/lazy/formatter.rs
use crate::table;

use super::LazyPlugin;

pub fn plugin() -> nvim_oxi::Result<LazyPlugin> {
    Ok(LazyPlugin::new("stevearc/conform.nvim").opts(table! {
        formatters_by_ft = table! {
            lua = ["stylua"],
            python = ["black"],
            javascript = ["prettier"],
            typescript = ["prettier"],
            json = ["prettier"],
            markdown = ["prettier"],
            rust = ["rustfmt"],
            yaml = ["prettier"],
            toml = ["taplo"]
        },
        format_on_save = table! {
            timeout_ms = 500,
            lsp_format = "fallback"
        }
    }))
}
